server/src/routes/api/v1/showsRouter.js

```diff
@@ -40,6 +40,43 @@ showsRouter.post("/", async (req, res) => {
   }
 })
 
+showsRouter.patch("/:id", async (req, res) => {
+  try {
+    const editedShow = cleanUserInput(req.body)
+    const showId = req.params.id
+    const editedFormattedShow = await Show.query().updateAndFetchById(showId, {
+      name: editedShow.name,
+      description: editedShow.description
+    })
+    const serializedShow = await ShowSerializer.getDetail(editedFormattedShow)
+    return res.status(200).json({ serializedShow })
+  } catch (error) {
+    if(error instanceof ValidationError) {
+      return res.status(422).json({ errors: error.data })
+    }
+    return res.status(500).json({ error })
+  }
+})
+
+showsRouter.delete("/:id", async (req, res) => {
+  try {
+    const showId = req.params.id
+    const show = await Show.query().findById(showId)
+    const reviews = await show.$relatedQuery("reviews")
+    for (const review of reviews) {
+      const votes = await review.$relatedQuery("votes")
+      for (const vote of votes) {
+        const deletedVote = await Vote.query().deleteById(vote.id)
+      }
+      const deletedReview = await Review.query().deleteById(review.id)
+    }
+    const deletedShow = await Show.query().deleteById(showId)
+    return res.status(200).json({deletedShow})
```

Comment:
`deletedShow` i believe will be just the number of shows deleted.

Further, this is going to execute a query for every single vote and review record, making a super costly set of operations. We can dramatically improve the efficiency of this:

```
await show.$relatedQuery("reviews").delete()
```
Of course, you'll also have to still iterate to delete related votes for each review. 

Theres something else though! You can update your table to have a `onDelete CASCADE`, but this would require updating your original migrations, so maybe put this off for a different app 

```
knex.schema.createTable('reviews', (table) => { 
  table.increments('id').primary() 
  table.integer('showId').unsigned() .
    .references('article.article_id') // 
    .onDelete('CASCADE') // If show is deleted, delete review as well.
})
```

Reply:
CASCADING looks cool, we'll definitely look to implement that the next time around